From 4a7015536e665bd8b21dc3c7b5884b7d94710d6b Mon Sep 17 00:00:00 2001
Message-Id: <4a7015536e665bd8b21dc3c7b5884b7d94710d6b.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Fri, 9 Jan 2015 15:55:15 +0530
Subject: [PATCH 89/95] FOR_UPSTREAM [VPG]: drm/i915: Add New RC6 Promotion
 timer for Media workloads to CHV

In normal cases, RC6 promotion timer is 1700us. This will result in
more time spent in C1 state. For more residency in C6 in case of
media workloads, this is changed to 250us. Not doing this for 3D workloads
as too many C6-C0 transition delays can result in performance impact.
Tracking the media workloads based on command submission to MFX engine

Issue: GMIN-3622
Change-Id: I353fee8a4d4720d3ca7eec705c4ad78eb08d7d68
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_lrc.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_lrc.c b/drivers/gpu/drm/i915/intel_lrc.c
index ee9fc6a..26bfbbe 100644
--- a/drivers/gpu/drm/i915/intel_lrc.c
+++ b/drivers/gpu/drm/i915/intel_lrc.c
@@ -1496,6 +1496,22 @@ int intel_execlists_submission_final(struct i915_execbuffer_params *params)
 
 	i915_gem_execbuffer_retire_commands(params->dev, params->file, ring, params->batch_obj);
 
+	/*
+	 * CHV: Extend RC6 promotion timer upon hitting Media workload to help
+	 * increase power savings with media scenarios.
+	 */
+	if (((params->args_flags & I915_EXEC_RING_MASK) == I915_EXEC_BSD) &&
+		IS_CHERRYVIEW(dev_priv->dev) && dev_priv->rps.enabled) {
+
+		vlv_modify_rc6_promotion_timer(dev_priv, true);
+
+		/* Start a timer for 1 sec to reset this value to original */
+		mod_delayed_work(dev_priv->wq,
+				&dev_priv->rps.vlv_media_timeout_work,
+				msecs_to_jiffies(1000));
+
+	}
+
 	return 0;
 
 error:
-- 
1.7.9.5

